### ISSUE TITLE
Does new release is planned soon?

### ISSUE THREAD

**Issue by stephane-klein** [golang-migrate/migrate#546]
Hi,

@dhui I see 30 commits between last `v4.14.0` release and `master` branch: https://github.com/golang-migrate/migrate/compare/v4.14.0...master

Have you planned to publish a new release soon?

Best regards,
Stéphane

**Comment by dhui** [comment]
Unfortunately release are blocked until [CI is able to properly create releases](https://github.com/golang-migrate/migrate/issues/252#issuecomment-767399329).
This [PR](https://github.com/golang-migrate/migrate/pull/521), which moves CI from CircleCI to Github Actions, is the most promising but I haven't had time to test it.

**Comment by stephane-klein** [comment]
cc @Its-Alex

**Comment by vallahaye** [comment]
Hello @dhui,

I have been following this topic for a while now and on several occasions you have said that you did not have time to test PR # 570. **Is there anything we can do to help you with this process?** Merging this PR seems to be the top priority as a lot of people (myself included) are relying on a new release. I plan to import `golang-migrate` as a library and the lack of support for `iofs` sources in the most recent release prevents me from using it.

Best regards,
Valentin

**Comment by dhui** [comment]
@vallahaye As you mentioned #570 is now the most promising PR for unblocking releases since it [uses GoReleaser](https://github.com/golang-migrate/migrate/issues/74) and Github Actions.

I still haven't found time to test it and can't make any promises as to when I will have time.

> Is there anything we can do to help you with this process? 

A co-maintainer would help. Ideally, someone who is experienced with Go and RDBMSs, or someone who's willing to learn and has the time to do so. A co-maintainer should expect to spend on average a couple hours a week on issues and larger tasks may take a few days.

For #570 specifically, someone would need to test releasing `migrate` on their own accounts (package cloud, docker hub, github), audit the github actions used for security (e.g. no tokens are leaked), ensure that all the proper binaries are created (with the correct source & db drivers) and that the artifacts are correctly pushed to the various repositories (package cloud, docker hub, github). If this all goes well, this should only take a couple hours, but could take half a day or more if there are issues. Unfortunately, I haven't been able to dedicate a full or half day to test #570

> I plan to import `golang-migrate` as a library and the lack of support for `iofs` sources in the most recent release prevents me from using it.

Use the master branch until the next release.

**Comment by Fontinalis** [comment]
Hey @dhui ,

I faced some issues today with the GitHub source driver and had to realize that the fix is already merged but the project is currently blocked from being able to do new releases. 

So I went ahead and tested `goreleaser` as mentioned as a potential replacement of the old CI. I started working from #570 , but had to most of the configurations to make it work, which makes me think that #570 doesn't include the right changes to be made. 

I tested it by creating a repo at ([package cloud](https://packagecloud.io/Fontinalis/migrate), [docker hub](https://hub.docker.com/r/rigglo/migrate), and in [my fork](https://github.com/Fontinalis/migrate) I also created releases to test and develop a ready to deploy solution. 

[The last release](https://github.com/Fontinalis/migrate/releases/tag/v4.14.2-alpha.6) is the one that I think is the closest to an ideal release, the process of creating a release is the following
- create a tag and push it (in a form of `v4.14.2`)
- GitHub Actions will run the release action, which will create the correct binaries and push the artifacts to the proper places (package cloud, docker hub) and create a draft release on github with the artifacts attached.
- Edit the draft release, add the changes notes and publish it.

Questions that I have
- Should I create a separate PR for this (since it's different than #570 ) and I can nicely include all the changes, details about it?

> A co-maintainer would help. Ideally, someone who is experienced with Go and RDBMSs, or someone who's willing to learn and has the time to do so. A co-maintainer should expect to spend on average a couple hours a week on issues and larger tasks may take a few days.

I'm happy to help. I've been using this project for a while now at work and I'm happy to give back in the form of maintaining it in a portion of my free time. 

Note: Also wrote up a changes note (it's in the last release in my fork), gathered all the changes since the last version which is.. WOW.. so many improvements!

**Comment by dhui** [comment]
Hi @Fontinalis, thanks so much for testing #570!

It looks promising!

> Should I create a separate PR for this

Yes please! 🙏 
I'd base your changes off of #570 so @johejo is properly attributed for their work.

Here are some comments to save you time before you open a PR:
* Why is `rigglo/migrate` used in `image_templates`? If it's purely for testing, let's go back to `migrate/migrate` in the PR.
* Why do we need `ruby/setup-ruby@v1`?
* The `Dockerfile` shouldn't be changed for this since it's used to create images locally. You can create a separate `Dockerfile` for GitHub Actions, similar to `Dockerfile.circleci`
* Keep the config for the other CI providers/hosts so we can run tests side-by-side. We can remove them later once everything is migrated
* Can you add support for arm for our docker images?

For now, we can run tests in CircleCI and release using GitHub Actions until our CI/tests are migrated to GitHub Actions.
I think GitHub Actions creating a draft release makes sense. Once tests are running in GitHub Actions and we're happy with the GoReleaser release notes template, we can automatically publish releases.

> I'm happy to help. I've been using this project for a while now at work and I'm happy to give back in the form of maintaining it in a portion of my free time.

Any help is appreciated! 🙏 
We'll need to setup a channel for discussions e.g. [GitHub Discussions](https://docs.github.com/en/discussions/quickstart) or [Gitter](https://gitter.im/) to help us be more consistent as maintainers.

> Note: Also wrote up a changes note (it's in the last release in my fork), gathered all the changes since the last version which is.. WOW.. so many improvements!

The changes LGTM at a quick glance. There is a breaking change with `postgres` where the advisory lock generation changed. Below is my note for it:

> When updating to this version of `migrate` with postgres, you need to ensure that no migrations are currently running and no old versions of migrate will run at the same time as new versions. The inputs for generating the advisory lock changed so this version will lock with a different key than previous versions.

**Comment by Fontinalis** [comment]
Awesome, will get right to it! Yes, will base from #570.

> Why is rigglo/migrate used in image_templates? If it's purely for testing, let's go back to migrate/migrate in the PR.

Yes, it was purely for testing, wanted to test the docker images as well, and to do so, needed a different repo in docker hub.

> Why do we need ruby/setup-ruby@v1?

It's needed to be able to do "gem install package_cloud". We could either do a "sudo gem install package_cloud" or do the official way by adding setting up ruby in that step. Otherwise the "gem install package_cloud" will return an error that it doesn't have permissions to a folder. [here is the exact error](https://github.com/Fontinalis/migrate/runs/3182834031?check_suite_focus=true#step:8:8)

> The Dockerfile shouldn't be changed for this since it's used to create images locally. You can create a separate Dockerfile for GitHub Actions, similar to Dockerfile.circleci

Yes, absolutely. Will create a Dockerfile.github or something like that for this.

Just a general note, some changes were made to make testing possible, like you mentioned changing the image name, or the package name at package cloud. 

> Can you add support for arm for our docker images?

Yes, will test it first then add it. 

> We'll need to setup a channel for discussions e.g. GitHub Discussions or Gitter to help us be more consistent as maintainers.

My vote is on GitHub Discussions, I like having most of the things in one place, and the GH discussions it would be in the repo and more interaction from the community could take place. At least that's my opinion.

**Comment by Fontinalis** [comment]
Just to track progress here as well: 
PR #604 was created and is waiting to be reviewed and approved.

**Comment by dhui** [comment]
All PR feedback is in #604 

> My vote is on GitHub Discussions, I like having most of the things in one place, and the GH discussions it would be in the repo and more interaction from the community could take place. At least that's my opinion.

Cool, I've enabled discussions!
